components/compiler-families/intel-compilers-devel/SOURCES/oneapi-modgen.lua

```diff
@@ -0,0 +1,937 @@
+-- to use oneAPI icc/icpc/ifort instead of gcc/g++/gfortran
+setenv("I_MPI_CC",   "icc")
+setenv("I_MPI_CXX",  "icpc")
+setenv("I_MPI_FC",   "ifort")
+setenv("I_MPI_F77",  "ifort")
+setenv("I_MPI_F90",  "ifort")
+
+setenv("MPI_DIR",    "]] .. PathCat(path, "mpi", version) .. [[")
+
+prepend_path("MODULEPATH", "]] .. C.ONEAPIMODS .. [[")
+prepend_path("MODULEPATH", "]] .. C.DEPPATH .. [[/intel-impi")
+
+load("mpi/" .. version)
+
+family("MPI")
+
+]]       )
+    end -- modtype="mpi"
+
+    -- Template for Intel(R) oneAPI MPI library for gnu/gnuXX
+    if modtype == "gnu-mpi" or modtype == "gnu" .. C.GNU .. "-mpi" then
+        local gnu = modtype:gsub("-mpi","")
+        return (header .. [[
+    help("\nThis module loads the Intel MPI environment " ..
+      "for use with ]] .. gnu .. [[\n" ..
+      "   mpif90   (Fortran source)\n" ..
+      "   mpicc    (C   source)\n" ..
+      "   mpicxx   (C++ source)\n" ..
+      "\nVersion " .. version .. "\n"
+    )
+    
+    whatis("Name: Intel MPI")
+    whatis("Version: " .. version)
+    whatis("Category: library, runtime support")
+    whatis("Description: Intel MPI Library (C/C++/Fortran for x86_64)")
+    whatis("URL: http://software.intel.com/en-us/articles/intel-mpi-library")
+    
+    -- Set wrappers to GNU compilers gcc/g++/gfortran
+    setenv("I_MPI_CC",   "gcc")
+    setenv("I_MPI_CXX",  "gxx")
+    setenv("I_MPI_FC",   "gfortran")
+    setenv("I_MPI_F77",  "gfortran")
+    setenv("I_MPI_F90",  "gfortran")
+    
+    setenv("MPI_DIR",    "]] .. PathCat(path, "mpi", version) .. [[")
+    
+    prepend_path("MODULEPATH", "]] .. C.ONEAPIMODS .. [[")
+    prepend_path("MODULEPATH", "]] .. C.DEPPATH .. "/" .. gnu .. [[-impi")
+    
+    load("mpi/" .. version)
+    
+    family("MPI")
+    
+    ]]   )
+    end -- modtype="gnu-mpi"
+
+    error("Function ModuleText called with undefined filetype.", 2)
+    os.exit(1)
+end
+
+-- Print command line help
+local function printhelp(isfull)
+    print("USAGE:\n  " .. arg[0] .. " COMMAND TYPE [arg] ")
+    if isfull then
+        print("\n  COMMANDs")
+        for k, v in pairs(Commands) do
+            if not v.ishidden then
+                print(string.format("  %-10s", k) .. v.help)
+            end
+        end
+        print("\n  TYPEs")
+        for k, v in pairs(ModTypes) do
+            print(string.format("  %-10s", k) .. v.help)
+        end
+    end
+    print()
+end
+
+-- Run md5sum against file; return the (first) checksum value
+local function md5sum(filename)
+    local output, iserror = RunOSCommand("md5sum -z " .. filename)
+    if iserror then
+        printformat("Unable to create md5sum for " .. filename, "error")
+    end
+    return output:sub(1, 32)
+end
+
+-- Find the default version
+local function find_default(path, isfullpath)
+    local linkinfo = lfs.symlinkattributes(PathCat(path, "default"))
+    if not linkinfo then
+        return nil
+    end
+    if not lfs.attributes(PathCat(path, linkinfo.target), "mode") then
```

Comment:
Apparently the "target" attribute was added in lfs 1.7.0, but EL8 only has lua-filesystem 1.6.3, so this fails for me.